lib/getPackagePaths.js

```diff
@@ -1,28 +1,38 @@
 const getManifest = require("./getManifest");
 const glob = require("./glob");
+const { join } = require("path");
 const { checker } = require("./blork");
+const { getPackagesSync } = require("@manypkg/get-packages");
 
 /**
- * Return array of package.json for Yarn workspaces.
+ * Return array of package.json for workspace packages.
  *
  * @param {string} cwd The current working directory where a package.json file can be found.
  * @param {string[]|null} ignorePackages (Optional) Packages to be ignored passed via cli.
  * @returns {string[]} An array of package.json files corresponding to the workspaces setting in package.json
  */
-function getWorkspacesYarn(cwd, ignorePackages = null) {
-	// Load package.json
-	const manifest = getManifest(`${cwd}/package.json`);
-
-	let packages = manifest.workspaces;
-	if (packages && packages.packages) {
-		packages = packages.packages;
+function getPackagePaths(cwd, ignorePackages = null) {
+	let packages;
+	// Ignore exceptions as we will rely on `getManifest` validation
+	try {
+		packages = getPackagesSync(cwd);
+	} catch (e) {
+		/**/
 	}
-
-	// Only continue if manifest.workspaces or manifest.workspaces.packages is an array of strings.
-	if (!checker("string[]+")(packages)) {
-		throw new TypeError("package.json: workspaces or workspaces.packages: Must be non-empty array of string");
+	packages = packages || {
+		tool: "root",
+		root: { dir: cwd },
+	};
+	packages.root.packageJson = getManifest(join(packages.root.dir, "package.json"));
+
+	if (packages.tool === "root") {
+		packages.packages = [];
```

Comment:
`packages.packages
` looks a bit confusing. Let's rename to `manypkg.packages` or smth like that?

Reply:
in renamed it to `workspace`